UnoCpp/DeckManager.cpp

```diff
@@ -1,9 +1,63 @@
 #include "DeckManager.h"
+#include "DeckData.h"
+#include "NumberCard.h"
+#include "JumpCard.h"
+#include "ReverseCard.h"
+#include "PlusTwoCard.h"
 
 void DeckManager::CreateDeck()
 {
+	std::vector<Enums::CardColor> colorList{ Enums::Blue, Enums::Green, Enums::Red, Enums::Yellow };
+
+	for (int i = 0; i < colorList.size(); i++)
+	{
+		Enums::CardColor cardColor = colorList[i];
+		CreateNumberCards(cardColor);
+		CreateJumpCards(cardColor);
+		CreateReverseCards(cardColor);
+		CreatePlusTwoCards(cardColor);
+	}
+}
+
+void DeckManager::CreateNumberCards(Enums::CardColor cardColor)
+{
+	for (int cardByColorIndex = 0; cardByColorIndex < DeckData::AMOUNT_OF_NORMAL_CARDS_BY_COLOR; cardByColorIndex++)
+	{
+		for (int numberIndex = 0; numberIndex < DeckData::NUMBER_RANGE_IN_CARDS; numberIndex++)
+		{
+			std::shared_ptr<NumberCard> card = std::make_shared<NumberCard>(card, cardColor, std::to_string(numberIndex));
+			_cards.push_back(card);
```

Comment:
There is another method to add to a vector without having to copying unnecessary things. I'd recommend to take a look at `emplace_back` and on this video: [optimization of vectors from the Cherno](https://www.youtube.com/watch?v=HcESuwmlHEY)